packages/fxa-settings/src/pages/Signin/index.tsx

```diff
@@ -362,7 +363,13 @@ const Signin = ({
               required
               autoFocus
               onChange={() => {
-                GleanMetrics.login.engage();
+                // Only log the engage event once. Note that this text box is autofocused, so
+                // using autofocus wouldn't be a good way to do this.
+                if (hasEngaged === false) {
+                  setHasEngaged(true);
+                  GleanMetrics.login.engage();
+                }
```

Comment:
Thought: this works fine because the state is for that in-memory session, and we don't have to consider the logout-login state in that case?

Reply:
Can you elaborate? What do you mean 'in-memory' session, and what is the logout-login case you are thinking of.

Reply:
Sorry for being unclear, the STR I was thinking of:
1. You focus and update the password.
2. The update state is called from `setHasEngaged`; metric recorded.
3. You sign-out.
4. Repeat step 1.
5. Step 2 metric wouldn't be recorded?

Reply:
Gotcha. Yeah the metric would be recorded again in step 5. I'm pretty sure this what we want.  What we don't want is for the metric to be emitted on every key press, which is what this bug specifically addresses.